include/envoy/ssl/tls_certificate_config.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Ssl {
+
+/**
+ * An instance of the TlsCertificateConfig
+ */
+class TlsCertificateConfig {
+public:
+  virtual ~TlsCertificateConfig() {}
+
+  /**
+   * @return a string of certificate chain
+   */
+  virtual const std::string& certificateChain() const PURE;
+
+  /**
+   * @return a string of private key
+   */
+  virtual const std::string& privateKey() const PURE;
+};
+
+typedef std::shared_ptr<TlsCertificateConfig> TlsCertificateConfigSharedPtr;
```

Comment:
Do we need shared pointers if we are only going to return const raw pointers? Can we make this unique_ptr?

Reply:
To support dynamic secret, secrets managed by secrets managed referenced by multiple components. 
It can be changed to unique_ptr but it needs to be change back to shared_ptr in the next PR.
I will update it to unique_ptr.